docs: adding content for module 1 (#6)
asyncapi/learning-paths: 100-level-AsyncAPI-Intro/intro.md

Eve is a programming genius renowned for her ability to solve various technical challenges. She lives at 7 Optimists Lane, Capuccinova.

The clock strikes 6 a.m. on a Monday, and the gentle notes of Lofi music gently rouse her from slumber.

After getting herself ready and brewing a cup of coffee, Eve casts a quick glance at the clock, only to realize that she's running behind schedule for her first work meeting with Guild. Guild happens to be one of the dedicated users of AsyncAPI, the organization where Eve acts as an open-source contributor!

### What is AsyncAPI?
AsyncAPI offers a standardized approach for describing asynchronous APIs and message-driven systems through YAML or JSON formats. The AsyncAPI specification makes it easier to work with diverse messaging protocols, message structures, and event-driven architectures.

> The AsyncAPI specification can be defined using an AsyncAPI document, but the document itself doesn't take any direct actions or enforce rules. The real magic of the specification materializes only when an AsyncAPI document is implemented or instantiated.

Reviewer: > The AsyncAPI specification can be defined using an AsyncAPI document,

That sentence sounds confusing to me. In fact, isn't the AsyncAPI document the one which is defined using the AsyncAPI specification 🤔 ? 
Maybe it's a language thing, not sure.

Reviewer: > when an AsyncAPI document is implemented or instantiated

What does it mean to be implemented/instantiated?
Sounds like it is something programmable or executable when it is not.
Maybe we wanted to mention that the key part is the tooling maybe?

Author: @smoya Editing to sentences as suggested to avoid confusion in such cases. Sound good now ?

> The AsyncAPI document is defined using the AsyncAPI specification, but the document itself doesn't take any direct actions or enforce rules. The real magic of the specification materializes when an AsyncAPI document is implemented using tooling(such as code generators).

Reviewer: 1. You don't implement an AsyncAPI document. You define it, later you use it.
2. Added " documentation rendering" to the list of tooling examples as it is most common)

```suggestion
The AsyncAPI document is defined using the AsyncAPI specification, but the document itself doesn't take any direct actions or enforce rules. The real magic of the specification materializes when an AsyncAPI document is used among tooling (such as documentation rendering or code generators).
```

Furthermore, the AsyncAPI ecosystem includes tools and code generators designed to take an AsyncAPI definition and automatically generate client and server code in multiple programming languages. Such functionality expedites development while significantly reducing potential errors when handling messages.

### Real-world use cases of AsyncAPI
Eve collaborates with Guild to address their logistics challenge using AsyncAPI. Guild, a global logistics giant operating in multiple countries with varying currencies, faces the complex task of calculating order logistics costs and route optimization for hundreds to thousands of requests, each with varying processing times.

> Eve introduces AsyncAPI to Guild, explaining how it can process their requests asynchronously, enabling simultaneous calculations (requests) without waiting for one request to finish before moving on to the next. Multiple requests are processed, and a result (response) is sent to the appropriate user once the calculation is performed.

Reviewer: > explaining how it can process their requests asynchronously

AsyncAPI is not software that you run and make things to work asynchronously.
The way AsyncAPI can help Guild is to share their API with clients, or by using it to autogenerate code, validate their incoming/outcoming messages, etc etc etc.

> Eve introduces AsyncAPI to Guild, explaining how AsyncAPI can help them share their asynchronous API with clients, validating their incoming messages and more, thereby enabling simultaneous calculations(requests) without waiting for one request to finish before moving on to the next. Multiple requests are processed, and a result (response) is sent to the appropriate user once the calculation is performed.

Reviewer: I tried to simplify and reduce redundancy a bit: 
WDYT?
```suggestion
Eve introduces AsyncAPI to Guild, explaining how AsyncAPI can help them share their asynchronous API with clients, validating their incoming messages and more, thereby enabling simultaneous calculations (requests) without waiting for one request to finish before moving on to the next. The result (response) is then sent to the appropriate user once the calculation is performed.
```

> Eve, a programming genius with a knack for solving diverse problems and helping others' lives at Optimists Lane 7, Capuccinova. 
> 
> It's 6 a.m. on a Monday, and pleasant Lofi music goes off, waking her up.
> 
> After getting ready and making her coffee, glancing at the clock showed her running late for her first work meeting with Adeo. Adeo is one of the users of AsyncAPI, the organisation that Eve works at as an Open-source contributor!

Reviewer 2: ```suggestion
Eve is a programming genius renowned for her ability to solve various technical challenges. She lives at 7 Optimists Lane, Capuccinova.

The clock strikes 6 a.m. on a Monday, and the gentle notes of Lofi music gently rouse her from slumber.

After getting herself ready and brewing a cup of coffee, Eve casts a quick glance at the clock, only to realize that she's running behind schedule for her first work meeting with Adeo. Adeo happens to be one of the dedicated users of AsyncAPI, the organization where Eve acts as an open-source contributor!
```

### What is AsyncAPI?

> AsyncAPI provides a standard way to describe asynchronous APIs and message-driven systems using YAML or JSON format, making it easier for developers to understand and work with different messaging protocols, message formats, and event-driven architectures. 
> AsyncAPI specification can be defined using an AsyncAPI document, but the document alone will not do or enforce anything. The specification does the magic only when an AsyncAPI document is implemented or instantiated.
> There are many tools and code generators that can take an AsyncAPI definition and automatically generate client and server code in various programming languages, saving time and reducing the potential for errors in message handling.

Reviewer 2: ```suggestion
AsyncAPI offers a standardized approach for describing asynchronous APIs and message-driven systems through YAML or JSON formats. The AsyncAPI specification makes it easier to work with diverse messaging protocols, message structures, and event-driven architectures.

The AsyncAPI specification can be defined using an AsyncAPI document, but the document itself doesn't take any direct actions or enforce rules. The real magic of the specification materializes only when an AsyncAPI document is implemented or instantiated.

Furthermore, the AsyncAPI ecosystem includes tools and code generators designed to take an AsyncAPI definition and automatically generate client and server code in multiple programming languages. Such functionality expedites development while significantly reducing potential errors when handling messages.
```

### Real-world use cases of AsyncAPI

> Eve works with Adeo to understand their problem and how AsyncAPI can help them solve it. 
> 
> Adeo explains their problem to Eve: Adeo is a massive logistics company spread across an entire planet consisting of various countries and currencies. As a logistics company worldwide, Adeo faces the challenge of calculating the logistics cost of a particular order from multiple sources and the difference in cost from order to order. They constantly run calculations to check which route is more cost-efficient or which yields them more profit per order. That could be 100s or 1000s of requests that need to be processed, each being processed at varying times, meaning one request would take 1 minute while another request might take 5 minutes.
> 
> Eve walks them through how AsyncAPI can help them by processing the requests asynchronously, meaning the calculations(requests) will be processed simultaneously without waiting for one request to complete before calculating another. This way, multiple requests are processed, and a result(response) is sent to the appropriate user/person once the calculation is performed. 
> 
> After helping Adeo with their problem, Eve moves on with her day. She serves herself with some chicken lasagna and sits down in her backyard to enjoy her meal as she watches the stars. 
> 
> Watching the stars, Eve notices what appears to be Haley's comment. But, she quickly realises it is not, as what seems to be a spaceship crash lands in her backyard.

Reviewer 2: ```suggestion
Eve collaborates with Adeo to address their logistics challenge using AsyncAPI. Adeo, a global logistics giant operating in multiple countries with varying currencies, faces the complex task of calculating order logistics costs and route optimization for hundreds to thousands of requests, each with varying processing times.

Eve introduces AsyncAPI to Adeo, explaining how it can process their requests asynchronously, enabling simultaneous calculations (requests) without waiting for one request to finish before moving on to the next. Multiple requests are processed, and a result (response) is sent to the appropriate user once the calculation is performed.

Having assisted Adeo, Eve enjoys her chicken lasagna in her backyard while stargazing. Amidst the stars, she spots what initially seems like Haley's Comet but soon realizes it's something else entirely. To her shock, a spaceship crash-lands in her backyard!
```
